ako-gateway-api/k8s/ako_init.go

```diff
@@ -1 +1,409 @@
+	var statusWG *sync.WaitGroup
+	//var leaderElectionWG *sync.WaitGroup
+	if len(waitGroupMap) > 0 {
+		// Fetch all the waitgroups
+		ingestionwg, _ = waitGroupMap[0]["ingestion"]
+		graphwg, _ = waitGroupMap[0]["graph"]
+		fastretrywg, _ = waitGroupMap[0]["fastretry"]
+		slowretrywg, _ = waitGroupMap[0]["slowretry"]
+		statusWG, _ = waitGroupMap[0]["status"]
+		//leaderElectionWG, _ = waitGroupMap[0]["leaderElection"]
+	}
+
+	/** Sequence:
+	  1. Initialize the graph layer queue.
+	  2. Do a full sync from main thread and publish all the models.
+	  3. Initialize the ingestion layer queue for partial sync.
+	  **/
+	// start the go routines draining the queues in various layers
+	var graphQueue *utils.WorkerQueue
+	// This is the first time initialization of the queue. For hostname based sharding, we don't want layer 2 to process the queue using multiple go routines.
+	retryQueueWorkers := uint32(1)
+	slowRetryQParams := utils.WorkerQueue{NumWorkers: retryQueueWorkers, WorkqueueName: lib.SLOW_RETRY_LAYER, SlowSyncTime: lib.SLOW_SYNC_TIME}
+	fastRetryQParams := utils.WorkerQueue{NumWorkers: retryQueueWorkers, WorkqueueName: lib.FAST_RETRY_LAYER}
+
+	numWorkers := uint32(1)
+	ingestionQueueParams := utils.WorkerQueue{NumWorkers: numWorkers, WorkqueueName: utils.ObjectIngestionLayer}
+	//numGraphWorkers := lib.GetshardSize()
+
+	numGraphWorkers := uint32(8)
+
+	graphQueueParams := utils.WorkerQueue{NumWorkers: numGraphWorkers, WorkqueueName: utils.GraphLayer}
+	statusQueueParams := utils.WorkerQueue{NumWorkers: numGraphWorkers, WorkqueueName: utils.StatusQueue}
+	graphQueue = utils.SharedWorkQueue(&ingestionQueueParams, &graphQueueParams, &slowRetryQParams, &fastRetryQParams, &statusQueueParams).GetQueueByName(utils.GraphLayer)
+
+	//err := PopulateCache()
+
+	// Setup and start event handlers for objects.
+	c.addIndexers()
+	c.Start(stopCh)
+
+	fullSyncInterval := os.Getenv(utils.FULL_SYNC_INTERVAL)
+	interval, err := strconv.ParseInt(fullSyncInterval, 10, 64)
+
+	// Set up the workers but don't start draining them.
+	if err != nil {
+		utils.AviLog.Errorf("Cannot convert full sync interval value to integer, pls correct the value and restart AKO. Error: %s", err)
+	} else {
+		// First boot sync
+		err = c.FullSyncK8s(false)
+		if err != nil {
+			// Something bad sync. We need to return and shutdown the API server
+			utils.AviLog.Errorf("Couldn't run full sync successfully on bootup, going to shutdown AKO")
+			lib.ShutdownApi()
+			return
+		}
+		if interval != 0 {
+			worker = utils.NewFullSyncThread(time.Duration(interval) * time.Second)
+			worker.SyncFunction = c.FullSync
+			worker.QuickSyncFunction = c.FullSyncK8s
+			go worker.Run()
+		} else {
+			utils.AviLog.Warnf("Full sync interval set to 0, will not run full sync")
+		}
+
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	//c.cleanupStaleVSes()
+
+	//leaderElector, err := utils.NewLeaderElector(informers.Cs, c.OnStartedLeading, c.OnStoppedLeading, c.OnNewLeader)
```

Comment:
this code can be removed. We need to revisit this once we finalize how to do the HA for this pod.

Reply:
removed comments